R/preprocessing.R

```diff
@@ -2429,20 +2445,57 @@ ReadVitessce <- function(
 #' @template note-reqdpkg
 #'
 ReadVizgen <- function(
-  data.dir,
-  transcripts = NULL,
-  spatial = NULL,
-  molecules = NULL,
-  type = 'segmentations',
-  mol.type = 'microns',
-  metadata = NULL,
-  filter = NA_character_,
-  z = 3L
+    data.dir,
+    transcripts = NULL,
+    spatial = NULL,
+    molecules = NULL,
+    type = 'segmentations',
+    mol.type = 'microns',
+    metadata = NULL,
+    filter = NA_character_,	
+    z = 3L,
+    use.BiocParallel = TRUE,
+    workers.total = 12,
+    DTthreads.pct = NULL,
+    coord.space = "micron", 
+    use.cellpose.out = TRUE
```

Comment:
Do you think we can infer this argument based on the presence of `cell_boundaries.parquet` files in `data.dir`? I want to make sure all of the older Vizgen outputs are still supported without requiring this parameter to be set.

Reply:
That should be possible.
Basically it will read `cellpose_micron_space.parquet `or `cellpose_mosaic_space.parquet `, else the older Vizgen output, right?

Reply:
Something like that, but my understanding was the cellpose outputs generated by Vizgen would are always named `cell_boundaries.parquet` based on their user guide: [https://vizgen.com/wp-content/uploads/2022/12/91600001_MERSCOPE-Instrument-User-Guide_Rev-F.pdf](https://vizgen.com/wp-content/uploads/2022/12/91600001_MERSCOPE-Instrument-User-Guide_Rev-F.pdf) (see page 45 and 46). Are your parquet output files generated by Vizgen, and if so, do you know what version of their software was used?

Reply:
Our data was generated by Vizgen, then we got the link to download them. I will check with them again but it's probably the `cell_boundaries.parquet` and `cell_by_gene.csv` and `cell_metadata.csv` files would be the official one.

Reply:
I will update the things we discussed here, plus some additional sanity checks are needed on `cell_boundaries.parquet`
see recent discussion #7080
thanks

Reply:
Thanks @alikhuseynov! Vizgen has also agreed to review this PR and make any necessary changes so users can handle cellpose segmentations

Reply:
Presence of a parquet file doesn't mean its a cellpose segmentation. Parquet files could be generated by either of the segmentation algorithms (see pg 29 of guide above).

Reply:
> Presence of a parquet file doesn't mean its a cellpose segmentation. Parquet files could be generated by either of the segmentation algorithms (see pg 29 of guide above).

yes, the idea is to allow users to load segmentation data (Cellpose or not) if `.parquet` is present. Some may have older data, ie `.hdf5` files.